Sources/Sentry/SentrySerialization.m

```diff
@@ -208,6 +208,12 @@ + (SentryEnvelope *_Nullable)envelopeWithData:(NSData *)data
             if (endOfEnvelope == i) {
                 i++; // 0 byte attachment
             }
+
+            if (data.length < (i + 1 + bodyLength)) {
+                SENTRY_LOG_DEBUG(@"Envelope is corrupted or has invalid data.")
```

Comment:
`m`: info on lengths (data.length, i, bodyLength) would probably be helpful in the log

Reply:
maybe also higher log level, similar to https://github.com/getsentry/sentry-cocoa/pull/4297/files#diff-b1a03b0a05f5b53fc2dcfa9bdeebe0d6285541a37b2eb9b93733b1c2a7e5af58R183